src/test/java/it/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMDescriptorIT.java

```diff
@@ -114,38 +106,29 @@ public void testFindProjectsCredentialsIdInvalid() throws Exception {
 
     @Test
     public void testFindProjectsCredentialsIdMissing() throws Exception {
-        BitbucketServerConfiguration bitbucketServerWithoutCredentials = new BitbucketServerConfiguration(
-                bbJenkinsRule.getBitbucketServerConfiguration().getAdminCredentialsId(),
-                bbJenkinsRule.getBitbucketServerConfiguration().getBaseUrl(), null, null);
-        bbJenkinsRule.addBitbucketServer(bitbucketServerWithoutCredentials);
-        HudsonResponse<List<JenkinsBitbucketProject>> response =
+        Response response =
                 RestAssured.expect()
-                        .statusCode(200)
+                        .statusCode(400)
```

Comment:
so this is interesting, 
While I don't think we need to change this as part of this test, but what about public repositories? You don't need credentials for them. I would _assume_ that public repositories is not fantastically common among our users but now it feels like we're forcing you to use credentials when you actually don't need to.

Reply:
Hmm, yes, I suppose that was always the case, except we're now more explicit about it?
I agree even if public repos aren't _common_ among our customers, we should still think about them. I'll dig a bit deeper. :+1:

Reply:
- [ ] make credentials optional for public repos?